Dfe.Data.SearchPrototype/Web/Tests/Unit/Models/ViewModels/Shared/PaginationTests.cs

```diff
@@ -0,0 +1,287 @@
+        {
+            // arrange
+            Pagination pagination = new(new Mock<IPager>().Object)
+            {
+                TotalRecordCount = 9,
+                RecordsPerPage = 10
+            };
+
+            // act
+            int result = pagination.TotalNumberOfPages;
+
+            // assert
+            result.Should().Be(1);
+        }
+
+        [Fact]
+        public void TotalNumberOfPages_WithTotalRecordCountZero_ThrowsArgumentException()
+        {
+            // arrange
+            Pagination pagination = new(new Mock<IPager>().Object)
+            {
+                TotalRecordCount = 0,
+                RecordsPerPage = 10
+            };
+
+            // act, assert
+            Action failedAction =
+                () => _ = pagination.TotalNumberOfPages;
+
+            ArgumentException exception = Assert.Throws<ArgumentException>(failedAction);
+
+            exception.Message.Should().Be("The record count must be greater than zero.");
+        }
+
+        [Fact]
+        public void TotalNumberOfPages_WithRecordsPerPageZero_ThrowsArgumentException()
+        {
+            // arrange
+            Pagination pagination = new(new Mock<IPager>().Object)
+            {
+                TotalRecordCount = 10,
+                RecordsPerPage = 0
+            };
+
+            // act, assert
+            Action failedAction =
+                () => _ = pagination.TotalNumberOfPages;
+
+            ArgumentException exception = Assert.Throws<ArgumentException>(failedAction);
+
+            exception.Message.Should().Be("The page size must be greater than zero.");
+        }
+
+        [Fact]
+        public void IsPageable_WithMoreThanOnePage_ReturnsTrue()
+        {
+            // arrange
+            Mock<IPager> mockPager = new();
+            mockPager.Setup(pager => pager.GetPageSequence(It.IsAny<int>(), It.IsAny<int>())).Returns([1, 2, 3, 4, 5]);
+
+            Pagination pagination = new(mockPager.Object)
+            {
+                TotalRecordCount = 20,
+                RecordsPerPage = 10
+            };
+
+            // act
+            bool result = pagination.IsPageable;
+
+            // assert
+            result.Should().BeTrue();
+        }
+
+        [Fact]
+        public void IsPageable_WithExactlyOnePage_ReturnsFalse()
+        {
+            // arrange
+            Mock<IPager> mockPager = new();
+            mockPager.Setup(pager => pager.GetPageSequence(It.IsAny<int>(), It.IsAny<int>())).Returns([1]);
+
+            Pagination pagination = new(mockPager.Object)
+            {
+                TotalRecordCount = 20,
+                RecordsPerPage = 10
+            };
+
+            // act
+            bool result = pagination.IsPageable;
+
+            // assert
+            result.Should().BeFalse();
+        }
+
+        [Fact]
+        public void IsPageable_WithMoreThanOnePage_ReturnsFalse()
```

Comment:
No it doesn't - incorrectly named test

Reply:
isPageable has been tested for true and false above i think this is just copied by mistake i removed it